### PR TITLE
Ingestor + projector cxi

### PR DESCRIPTION
**Pull request by JulReinhardt**

### CODE REVIEW

**Comment by JulReinhardt** [comment]
If either of you could have a look at the cxi ingestor and make comments, I'd appreciate a discussion about:

- [ ] how to handle multiple data sets to be compared (as one --> rescaling pixelsizes, sorting energies etc. probably not done in the ingestor)
- [ ] naming conventions for projections
- [ ] naming of the frame_data_keys since 'raw' is not actually true for reconstructed ptychographic images

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,6 @@
 *.tif
 build/
 dist/
-*.egg-info/
\ No newline at end of file
+*.egg-info/
+.vscode
+venv
```
Comment (dylanmcreynolds) [inline .gitignore]: Doesn't this need to be 'venv/'?

File: xicam/spectral/ingestors/NXcxi_ptycho.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    rec_stack = []
+    energy_stack = []
+    energy_eV_stack = []
+    wavelength_stack = []
+
+    rec_shape = h5['entry_1']['image_1']['data'][()].shape
+    dim_x = rec_shape[0]
+    dim_y = rec_shape[1]
+
+    def rec_psize_nm(energy, corner_x, corner_y, corner_z):
```
Comment (dylanmcreynolds) [inline xicam/spectral/ingestors/NXcxi_ptycho.py]: Is there a reason this is a nested function?

**Review by dylanmcreynolds — COMMENTED** [review]
I have a sample pytest for an ingestor that you might consider looking at for the pystxm ingestor. It's not released, but can be found: https://github.com/als-computing/beamline-data/blob/mapped_ingestor/docstream/tests/test_docstream.py

File: .gitignore (hunk 1 above)
Comment (JulReinhardt) [reply .gitignore]: probably yes, don't how that sneaked in there

File: xicam/spectral/ingestors/NXcxi_ptycho.py (hunk 2 above)
Comment (JulReinhardt) [reply xicam/spectral/ingestors/NXcxi_ptycho.py]: Probably not, only my sloppy writing ...

**Comment by JulReinhardt** [comment]
> These entry points look odd to me. If we're registering them xi-cam-wide, how do we map from one person's application/x-hdf5 or x-fits to another's idea of what those content types mean? I assume some code somewhere is mapping file extension to content type? Or am I completely misreading that?

I don't know, that was one of my questions that still needs to be addressed. @ronpandolfi can you comment, when you have a moment for this?

**Comment by ronpandolfi** [comment]
> These entry points look odd to me. If we're registering them xi-cam-wide, how do we map from one person's application/x-hdf5 or x-fits to another's idea of what those content types mean? I assume some code somewhere is mapping file extension to content type? Or am I completely misreading that?

Its an unresolved issue that there's no good way to dispatch to different ingestors associated with the same mimetype. We've had some ideas, but none that I'm fully satisfied with yet.

**Review by ronpandolfi — APPROVED** [review]

**Review by ronpandolfi — APPROVED** [review]

**Pull request merged**